zipkin-storage/zipkin2_cassandra/src/main/resources/zipkin2-schema-indexes.cql

```diff
@@ -33,12 +33,14 @@ CREATE TABLE IF NOT EXISTS zipkin2.span_by_service (
     AND speculative_retry = '95percentile'
     AND comment = 'Secondary table for looking up span names by a service name.';
 
+DROP INDEX IF EXISTS zipkin2.span_annotation_query_idx;
```

Comment:
was this there before?

Reply:
Only if the user has already used ` zipkin-storage/zipkin2_cassandra` and created the schema.
And if that is the case then the code won't actually be calling this cql file.

So this line only has any purpose if the cql file is going to be manually run.

This comes back to this question…
> upgrade schema approach, or require users to drop keyspace if already using this storage backend

If we're happy to say anyone already using the new schema is expected to either:
 - manually run `cqlsh -f zipkin2-schema-indexes.cql`, or
 - drop keyspace and let it be recreated from scratch.

Then this is approach is ok.
If not then we need something more along the lines of `zipkin-storage/cassandra/src/main/resources/cassandra-schema-cql3-upgrade-1.txt`